android/src/main/java/com/mapbox/mapboxgl/GlobalMethodHandler.java

```diff
@@ -36,6 +36,19 @@ public void onMethodCall(MethodCall methodCall, MethodChannel.Result result) {
                 installOfflineMapTiles(tilesDb);
                 result.success(null);
                 break;
+            case "downloadOfflineRegion":
+                //Get download region arguments from caller
+                Gson gson = new Gson();
+                OfflineRegionData args = gson.fromJson(methodCall.arguments.toString(), OfflineRegionData.class);
+                //Start downloading
+                OfflineManagerUtils.downloadRegion(args, result, registrar, args.getAccessToken());
```

Comment:
@kleeb I just tried it again, but it fails for me on Android because the access token is missing in the http request. Could this be, because gson can't set the access token in `OfflineRegionData` as there's no setter/constructor parameter for `accessToken`? I think the better way forward might be removing the `accessToken` member from `OfflineRegionData` and directly looking the token up from the arguments in this line?